tests/pipeline_composite_functions_test.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import unittest
+from dataclasses import dataclass
+from typing import List, Set
+
+import pyspark
+from absl.testing import parameterized
+
+import pipeline_dp
+from pipeline_dp import pipeline_composite_functions as composite_funcs
+
+
+def _materialize_col(backend: pipeline_dp.PipelineBackend, col):
+    if isinstance(backend, pipeline_dp.SparkRDDBackend):
+        return col.collect()
+    else:
+        return list(col)
+
+
+def _create_platform_supported_backends(backends_in_scope: Set[str]):
+    result = set()
+    for backend in backends_in_scope:
+        if backend == "local":
+            result.add(pipeline_dp.LocalBackend())
+        elif backend == "beam":
+            result.add(pipeline_dp.BeamBackend())
+        elif backend == "spark":
+            if sys.version_info.minor > 7 or sys.version_info.major != 3:
+                # if python3 <= 3.7 then there are serialization problems.
+                result.add(
+                    pipeline_dp.SparkRDDBackend(
+                        pyspark.SparkContext.getOrCreate(pyspark.SparkConf())))
+        elif backend == "multi_proc_local":
+            if sys.platform != 'win32' and sys.platform != 'darwin':
+                result.add(
+                    pipeline_dp.pipeline_backend.MultiProcLocalBackend(
+                        n_jobs=1))
+    return result
+
+
+_ALL_BACKENDS = {"local", "beam", "spark", "multi_proc_local"}
+
+
+# TODO: check whether possible to move it inside.
+@dataclass
+class TestContainer:
+    x: int
+    y: str
+    z: List[str]
+
+
+class PipelineCompositeFunctionsTest(parameterized.TestCase):
```

Comment:
@dvadym , do you have idea why tests in this file in python 3.8 and 3.9 on Ubuntu might fail for RDD with errors like this: https://paste.googleplex.com/5441345786216448?

Reply:
I checked Spark tests and recalled that there was some problem with creating Spark context multiple times. Could you please try to create it only once? [Here](https://github.com/OpenMined/PipelineDP/blob/be9087db6d7a7ab7aaa13aed7aacaa052e28686e/tests/private_spark_test.py#L35) an example

Reply:
Rewrote the test file, now we have separate test classes for each backend, the same way pipeline_backend_test.py is written and these tests are not parametrized by backend anymore. I think this way is better because it is consistent with pipeline_backend_test.py and also more maintainable.

Reply:
Now it works, probably we had to call SparkContext.parallelize(input_collection).